enterprise/frontend/src/metabase-enterprise/tenants/components/TenantCollectionPermissionsPage/TenantCollectionPermissionsPage.tsx

```diff
@@ -0,0 +1,152 @@
+import { useCallback, useEffect } from "react";
+import type { Route } from "react-router";
+import { push } from "react-router-redux";
+import { t } from "ttag";
+import _ from "underscore";
+
+import { CollectionPermissionsHelp } from "metabase/admin/permissions/components/CollectionPermissionsHelp";
+import {
+  PermissionsEditor,
+  PermissionsEditorEmptyState,
+} from "metabase/admin/permissions/components/PermissionsEditor";
+import { PermissionsPageLayout } from "metabase/admin/permissions/components/PermissionsPageLayout";
+import { PermissionsSidebar } from "metabase/admin/permissions/components/PermissionsSidebar";
+import {
+  initializeTenantCollectionPermissions,
+  loadTenantCollectionPermissions,
+  saveTenantCollectionPermissions,
+  updateTenantCollectionPermission,
+} from "metabase/admin/permissions/permissions";
+import type {
+  CollectionIdProps,
+  CollectionPermissionEditorType,
+  CollectionSidebarType,
+} from "metabase/admin/permissions/selectors/collection-permissions";
+import Collections from "metabase/entities/collections";
+import Groups from "metabase/entities/groups";
+import { connect } from "metabase/lib/redux";
+import type { Collection, CollectionId, GroupId } from "metabase-types/api";
+import type { State } from "metabase-types/store";
+
+import {
+  getIsTenantDirty,
+  getTenantCollectionEntity,
+  getTenantCollectionsPermissionEditor,
+  getTenantCollectionsSidebar,
+  tenantCollectionsQuery,
+} from "./selectors";
+
+const mapDispatchToProps = {
+  initialize: initializeTenantCollectionPermissions,
+  loadPermissions: loadTenantCollectionPermissions,
+  navigateToItem: ({ id }: { id: CollectionId }) =>
+    push(`/admin/permissions/tenant-collections/${id}`),
+  updateCollectionPermission: updateTenantCollectionPermission,
+  savePermissions: saveTenantCollectionPermissions,
+};
+
+const mapStateToProps = (state: State, props: CollectionIdProps) => {
+  return {
+    sidebar: getTenantCollectionsSidebar(state, props),
+    permissionEditor: getTenantCollectionsPermissionEditor(state, props),
+    isDirty: getIsTenantDirty(state),
+    collection: getTenantCollectionEntity(state, props),
+  };
+};
+
+type UpdateCollectionPermissionParams = {
+  groupId: GroupId;
+  collection: Collection;
+  value: unknown;
+  shouldPropagate: boolean;
+};
+
+type TenantCollectionPermissionsPageProps = {
+  params: CollectionIdProps["params"];
+  sidebar: CollectionSidebarType;
+  permissionEditor: CollectionPermissionEditorType;
+  collection: Collection;
+  navigateToItem: (item: { id: CollectionId }) => void;
+  updateCollectionPermission: ({
+    groupId,
+    collection,
+    value,
+    shouldPropagate,
+  }: UpdateCollectionPermissionParams) => void;
+  isDirty: boolean;
+  savePermissions: () => void;
+  loadPermissions: () => void;
+  initialize: () => void;
+  route: Route;
+};
+
+function TenantCollectionPermissionsPageView({
```

Comment:
do we really need a whole new page for this? couldn't we leverage the existing page and pass it some props? or is the logic really that different?

Reply:
I can make it use the existing page with props, but I wasn't sure where standard line is for adding props vs combining the same components in different ways as a new component is in the FE.

Reply:
yeah that's a familiar story, permissions are pretty redux-spaghetti-heavy. thanks for looking at it.

taking a holistic approach to untangling permissions a bit is probably the better course